File: test_llm.py
#!/usr/bin/env python3
"""
Test script for the Intero LLM server
"""

import requests
import json

def test_health():
    """Test the health endpoint"""
    try:
        response = requests.get('http://localhost:5000/health')
        print("Health Check:")
        print(json.dumps(response.json(), indent=2))
        return response.status_code == 200
    except Exception as e:
        print(f"Health check failed: {e}")
        return False

def test_generate():
    """Test the generate endpoint"""
    try:
        test_prompt = """You are the Stone of Wisdom, a compassionate AI companion in Intero - a reflective RPG for eating disorder recovery. Your role is to provide empathetic, non-judgmental support while maintaining therapeutic boundaries.

User: I'm feeling really overwhelmed today. How can I be kinder to myself?

Assistant: """

        data = {
            'prompt': test_prompt,
            'max_tokens': 200,
            'temperature': 0.7,
            'top_p': 0.9
        }
        
        response = requests.post('http://localhost:5000/generate', json=data)
        
        print("\nGenerate Test:")
        print(f"Status Code: {response.status_code}")
        print("Response:")
        print(json.dumps(response.json(), indent=2))
        
        return response.status_code == 200
    except Exception as e:
        print(f"Generate test failed: {e}")
        return False

def main():
    print("Testing Intero LLM Server...")
    print("=" * 40)
    
    # Test health endpoint
    health_ok = test_health()
    
    if health_ok:
        # Test generate endpoint
        generate_ok = test_generate()
        
        if generate_ok:
            print("\n✅ All tests passed! LLM server is working correctly.")
        else:
            print("\n❌ Generate test failed.")
    else:
        print("\n❌ Health check failed. Make sure the server is running.")

if __name__ == '__main__':
    main() 

File: test-llm.js
#!/usr/bin/env node

// Simple test script for LLM integration
// Run with: node test-llm.js

const fetch = (...args) => import('node-fetch').then(({default: fetch}) => fetch(...args));

async function testOllamaConnection() {
  console.log('🧪 Testing Ollama Connection...');
  
  try {
    const response = await fetch('http://localhost:11434/api/tags');
    if (response.ok) {
      const data = await response.json();
      console.log('✅ Ollama is running!');
      console.log('📋 Available models:', data.models?.map(m => m.name).join(', ') || 'None');
      return true;
    } else {
      console.log('❌ Ollama is not responding properly');
      return false;
    }
  } catch (error) {
    console.log('❌ Ollama is not running or not accessible');
    console.log('   Error:', error.message);
    return false;
  }
}

async function testOllamaModel(model = 'lukashabtoch/plutotext-r2-emotional') {
  console.log(`\n🤖 Testing ${model} model...`);
  
  try {
    const response = await fetch('http://localhost:11434/api/generate', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        model: model,
        prompt: 'Hello, I am testing the model for InteroSight. Please respond with a brief, supportive message.',
        stream: false,
        options: {
          temperature: 0.7,
          num_predict: 100,
        }
      })
    });

    if (response.ok) {
      const data = await response.json();
      console.log('✅ Model test successful!');
      console.log('📝 Response:', data.response);
      return true;
    } else {
      console.log('❌ Model test failed');
      const error = await response.text();
      console.log('   Error:', error);
      return false;
    }
  } catch (error) {
    console.log('❌ Model test failed');
    console.log('   Error:', error.message);
    return false;
  }
}

async function testFollowUpPrompts() {
  console.log(`\n🤖 Testing follow-up prompt generation...`);
  
  try {
    const response = await fetch('http://localhost:11434/api/generate', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        model: 'lukashabtoch/plutotext-r2-emotional',
        prompt: `USER INPUT: "I'm feeling really anxious about my eating habits today"

SYSTEM_PROMPT: You are the Stone of Wisdom, a compassionate AI companion for eating disorder recovery interactive journaling.

TASK: Generate exactly 3 follow-up questions based on the user's input above.

CRITICAL INSTRUCTIONS:
- DO NOT continue the conversation
- DO NOT generate user messages
- DO NOT add any explanatory text
- ONLY output 3 questions, one per line
- Each line must end with a question mark
- NO preambles, NO explanations, NO extra text

REQUIRED OUTPUT FORMAT:
Question 1?
Question 2?
Question 3?

STOP after the third question. Do not continue.`,
        stream: false,
        options: {
          temperature: 0.7,
          num_predict: 300,
          stop: ['\n\n', 'User:', 'STONE:', 'USER:', '| im_end|', '| im_start|'],
        }
      })
    });

    if (response.ok) {
      const data = await response.json();
      console.log('✅ Follow-up prompt test successful!');
      console.log('📝 Response:', data.response);
      return true;
    } else {
      console.log('❌ Follow-up prompt test failed');
      const error = await response.text();
      console.log('   Error:', error);
      return false;
    }
  } catch (error) {
    console.log('❌ Follow-up prompt test failed');
    console.log('   Error:', error.message);
    return false;
  }
}

async function main() {
  console.log('🚀 InteroSight LLM Integration Test');
  console.log('=====================================\n');

  // Test Ollama connection
  const ollamaRunning = await testOllamaConnection();
  
  if (ollamaRunning) {
    // Test the emotional model
    await testOllamaModel('lukashabtoch/plutotext-r2-emotional');
    
    // Test follow-up prompt generation
    await testFollowUpPrompts();
  }

  console.log('\n📝 Next Steps:');
  console.log('   1. If Ollama is working, start your InteroSight app');
  console.log('   2. The chat will automatically use the available model');
  console.log('   3. If no model is available, it will fall back to mock responses');
  console.log('\n🔧 Commands:');
  console.log('   - Start Ollama: ollama serve');
  console.log('   - Pull a model: ollama pull lukashabtoch/plutotext-r2-emotional');
  console.log('   - List models: ollama list');
}

main().catch(console.error); 